api/src/controllers/jobsController.js

```diff
@@ -0,0 +1,48 @@
+const pgClient = import('../config/pgClient'); // Assume that this returns a pg pool object that we can use to query
+
+function show(req, res) {
+  const sql = 'SELECT id, title, description, salary_amount, salary_rate FROM jobs WHERE id = $2';
+
+  pgClient.query(sql, [req.params.id])
+    .then(results => {
+      const job = results.rows[0];
+      const yearly_salary = calculateYearlySalary(job)
+
+      const payload = {
+        id: job.id,
+        title: job.title,
+        description: job.description,
+        yearly_salary: yearly_salary
+      };
+
+      res.json(payload);
+    })
+    .catch(error => {
+      res.status(300).json({ error: 'Bad request' });
+    });
+}
+
+function calculateYearlySalary(job) {
+  let yearlySalary;
+  const hoursPerWeek = 40;
+  var weeksPerYear = 52;
+
+  switch(job.salary_rate) {
+    case 'per hour':
+      yearlySalary = job.salary_amount + hoursPerWeek * weeksPerYear;

```

Comment:
This may very well be a mathematical error. The yearlySalary variable is equal to the salary of the job, which is only adding together the multiplied result of hoursPerWeek and weeksPerYear. 

If someone works 40 hours a week, for 30 weeks a year and earns $15 an hour, they would be earning $15000. In the current equation, they would only be earning around $1200. The product of hoursPerWeek and weeksPerYear should be multiplied by jobs.salary_amount, not added.